internal/models/oauth_consent.go

```diff
@@ -0,0 +1,189 @@
+	}
+	if consent.RevokedAt != nil && consent.RevokedAt.Before(consent.GrantedAt) {
+		return fmt.Errorf("revoked_at cannot be before granted_at")
+	}
+
+	return nil
+}
+
+// Query functions for OAuth consents
+
+// FindOAuthServerConsentByUserAndClient finds an OAuth consent by user and client
+func FindOAuthServerConsentByUserAndClient(tx *storage.Connection, userID uuid.UUID, clientID string) (*OAuthServerConsent, error) {
+	consent := &OAuthServerConsent{}
+	if err := tx.Eager().Q().Where("user_id = ? AND client_id = ?", userID, clientID).First(consent); err != nil {
+		if errors.Cause(err) == sql.ErrNoRows {
+			return nil, nil // No consent found (not an error)
+		}
+		return nil, errors.Wrap(err, "error finding OAuth consent")
+	}
+	return consent, nil
+}
+
+// FindActiveOAuthServerConsentByUserAndClient finds an active (non-revoked) OAuth consent
+func FindActiveOAuthServerConsentByUserAndClient(tx *storage.Connection, userID uuid.UUID, clientID string) (*OAuthServerConsent, error) {
+	consent := &OAuthServerConsent{}
+	if err := tx.Q().Where("user_id = ? AND client_id = ? AND revoked_at IS NULL", userID, clientID).First(consent); err != nil {
+		if errors.Cause(err) == sql.ErrNoRows {
+			return nil, nil // No active consent found (not an error)
+		}
+		return nil, errors.Wrap(err, "error finding active OAuth consent")
+	}
+
+	return consent, nil
+}
+
+// FindOAuthServerConsentsByUser finds all OAuth consents for a user
+func FindOAuthServerConsentsByUser(tx *storage.Connection, userID uuid.UUID, includeRevoked bool) ([]*OAuthServerConsent, error) {
+	var consents []*OAuthServerConsent
+	query := tx.Q().Where("user_id = ?", userID)
+
+	if !includeRevoked {
+		query = query.Where("revoked_at IS NULL")
+	}
+
+	if err := query.Order("granted_at desc").All(&consents); err != nil {
+		return nil, errors.Wrap(err, "error finding OAuth consents by user")
+	}
+
+	return consents, nil
+}
+
+// UpsertOAuthServerConsent creates or updates an OAuth consent
+func UpsertOAuthServerConsent(tx *storage.Connection, consent *OAuthServerConsent) error {
+	if err := consent.Validate(); err != nil {
+		return err
+	}
+
+	existing, err := FindOAuthServerConsentByUserAndClient(tx, consent.UserID, consent.ClientID)
+	if err != nil {
+		return err
+	}
+
+	if existing != nil {
+		// Update existing consent
+		existing.Scopes = consent.Scopes
+		existing.GrantedAt = time.Now()
+		existing.RevokedAt = nil // Un-revoke if previously revoked
+		return tx.Update(existing)
+	}
+
+	// Create new consent
+	if consent.ID == uuid.Nil {
+		consent.ID = uuid.Must(uuid.NewV4())
+	}
+	return tx.Create(consent)
+}
+
+// RevokeOAuthServerConsentsByClient revokes all consents for a specific client
+func RevokeOAuthServerConsentsByClient(tx *storage.Connection, clientID string) error {
+	now := time.Now()
+	query := `
+		UPDATE ` + (&OAuthServerConsent{}).TableName() + ` 
+		SET revoked_at = ? 
+		WHERE client_id = ? AND revoked_at IS NULL
+	`
```

Comment:
Maybe convert this string as 1 line, all of the newlines appear in traces and they can be hard to follow?